src/dispatch.c

```diff
@@ -82,3 +83,452 @@ natsStatus natsSub_enqueueUserMessage(natsSubscription *sub, natsMsg *msg)
     return NATS_OK;
 }
 
+static inline void _removeFromDispatchQueue(natsDispatcher *d, natsMsg *msg)
+{
+    d->queue.head = msg->next;
+    if (d->queue.tail == msg)
+        d->queue.tail = NULL;
+    msg->next = NULL;
```

Comment:
The name of the function is not really indicative that this message needs to be the head of the list. I understand that we have selected the head before calling this function, and so this is just updating the list, but the fact that this function is named "remove" and we pass the message, one could think that the function is going to remove a message anywhere in the queue.